### DISCUSSION TITLE
Pagination : Infinite Scroll

### DISCUSSION THREAD

**Discussion by ijaspreetsinghh** [supabase/supabase#6753] [Questions] [answered]
Hi, How can we implement pagination with supabase and flutter, like when scrolled down 80% in a listview load next page, without knowing how many pages we have. can this be done without playing with limits?

I am trying to use range(start, end), but this is not working, every time, it's returning the same x rows, though it is supposed to return new rows with an updated range index.

like range(0,5), for first 6 rows, and range(6,106) for next 100 rows. it's the same for both. both times first 6 rows are being returned.

 (289, 288, 287, 286, 285, 284) these are the ids for first 6 rows to be fetched, for next range, it is supposed to be 100 next rows, (284,283,. . . . . . . .  184 ).

![image](https://user-images.githubusercontent.com/44422128/167593113-5a229002-991c-418a-b20e-dfbb718dec46.png)

Seems like It always pick first 6 rows: 

```dart
     final response = await supabaseClient.from('Sales').select(''' 
     *,Customer:customer_id(id,last_name,first_name),Users:user(token,name,email).
     .filter('location_id', 'eq', appLocationController.currentLocation.value.id)
     .filter('created_at','gte',DateTime(start_date.year,start_date.month,start_date.day,0,0,0,0,0,).toIso8601String(),)
     .filter('created_at','lte',DateTime(end_date.year,end_date.month,end_date.day,23,59,59,999,9999,).toIso8601String(),)
     .range(range_from,range_to,)
     .order('created_at',).execute(count: CountOption.exact)
```

**Comment by dshukertjr** [comment, marked as answer]
@ijaspreetsinghh Sorry that you are experience this. I have opened an issue [here](https://github.com/supabase-community/supabase-flutter/issues/119) so that you can track the fix of this issue.

**Reply by ijaspreetsinghh** [reply]
Hi thanks for the opening issue.
Problem was solved with the help following function.

RangeSize getPagination({ 
   required int page, 
 }) { 
   final from = page * default_range; 
   final to = from + default_range - 1; 
  
   return RangeSize(from: from, to: to); 
 }

and make sure to parse the new response body. In my case i was parsing the initial response, not new response which was loaded every time I scrolled.

**Reply by dshukertjr** [reply]
@ijaspreetsinghh Glad to her that you are no longer having this issue!
